x/ccv/consumer/module.go

```diff
@@ -92,16 +93,22 @@ func (AppModuleBasic) GetQueryCmd() *cobra.Command {
 	return cli.NewQueryCmd()
 }
 
+type StakingKeeper interface {
+	GetAllValidators(ctx sdk.Context) (validators []stakingtypes.Validator)
+}
+
 // AppModule represents the AppModule for this module
 type AppModule struct {
 	AppModuleBasic
 	keeper keeper.Keeper
+	sk     StakingKeeper
 }
 
 // NewAppModule creates a new consumer module
-func NewAppModule(k keeper.Keeper) AppModule {
+func NewAppModule(k keeper.Keeper, sk StakingKeeper) AppModule {
```

Comment:
Doesn't the consumer keeper act as the staking keeper? An explanation of why we pass in a staking keeper here would be nice

Reply:
staking keeper is used to get existing validators on sovereign staking, to remove voting power.

Reply:
Imo it would be more understandable to make a new file, `x/ccv/democracy/module.go` which defines a different `NewAppModule` constructor which accepts a staking keeper. Then we wouldn't have to pass in nil for the normal consumer: https://github.com/Stride-Labs/interchain-security/blob/67aa39b7fa795abb1519a5afb9731fdfa34d0c71/app/consumer/app.go#L348

Reply:
In general, I'm not the most familiar with the differences between consumer and democracy consumer, see https://github.com/cosmos/interchain-security/issues/680, so maybe my comments here are not relevant

Reply:
Consumer is for the chain that directly launch consumer chain from genesis, and democracy consumer is for the chain upgrading to consumer chain from sovereign I think.

Reply:
> Imo it would be more understandable to make a new file, `x/ccv/democracy/module.go` which defines a different `NewAppModule` constructor which accepts a staking keeper. Then we wouldn't have to pass in nil for the normal consumer:

This would require adding lots of duplicated code probably.